packages/clay-core/src/table/Column.tsx

```diff
@@ -56,29 +74,86 @@ export const Column = React.forwardRef<HTMLTableCellElement, Props>(
 			className,
 			delimiter,
 			expanded,
+			keyValue,
+			sorting,
 			textAlign,
+			textValue,
 			truncate,
 			wrap = true,
 			...otherProps
 		},
 		ref
 	) {
+		const {onSortChange, sort, sortDescriptionId} = useTable();
 		const scope = useScope();
 		const As = scope === Scope.Head ? 'th' : 'td';
 
 		return (
 			<As
 				{...otherProps}
+				aria-describedby={
+					scope === Scope.Head && sorting
+						? sortDescriptionId
+						: undefined
+				}
+				aria-sort={
+					scope === Scope.Head && sorting
+						? sort && keyValue === sort.column
+							? sort.direction
+							: 'none'
+						: undefined
+				}
 				className={classNames(className, {
 					'table-cell-expand': truncate || expanded,
 					[`table-cell-${delimiter}`]: delimiter,
 					[`table-column-text-${textAlign}`]: textAlign,
 					[`text-${align}`]: align,
 					'table-cell-ws-nowrap': !wrap,
+					'table-head-title': scope === Scope.Head,
 				})}
 				ref={ref}
 			>
-				{truncate ? (
+				{scope === Scope.Head && sorting ? (
+					<a
+						aria-describedby={
+							scope === Scope.Head && sorting
+								? sortDescriptionId
+								: undefined
+						}
+						className="inline-item text-truncate-inline"
+						href="#"
+						onClick={(event) => {
+							event.preventDefault();
+							onSortChange(
+								{
+									column: keyValue!,
+									direction:
+										sort && keyValue === sort.column
+											? sort.direction === 'ascending'
+												? 'descending'
+												: 'ascending'
+											: 'ascending',
+								},
+								textValue!
+							);
+						}}
+						role="button"
+					>
+						<span className="text-truncate">{children}</span>
+
+						{sort && keyValue === sort.column && (
```

Comment:
This looks like it matches the current Lexicon table designs, but in the future we will likely need to improve this so that there is a visual indicator that shows which columns allow for sorting and which do not. So no change needed for now, but thought it worth mentioning.

Reply:
Yeah, this actually seems a bit confusing, I think the only indication that Lexicon adds is when you hover over the header column, change the color and add the underline but there really isn't anything when the component is viewed without interacting with he.